docgenerator/src/main/scala/codacy/codenarc/docs/DocGenerator.scala

```diff
@@ -0,0 +1,316 @@
+
+    val regexPattern = ruleInfoRegex(ruleName)
+    val regexMatch = regexPattern.findFirstIn(ruleInfoMarkdown).getOrElse("")
+    regexMatch.substring(regexMatch.indexOf(System.lineSeparator) + 1) // remove the title
+  }
+
+  /**
+    * Get rule simplified description from codenarc-base-messages.properties file
+    *
+    * @param directory CodeNarc base directory
+    * @param patternId rule/pattern name
+    * @return Simplified Description
+    */
+  def getRuleBasicDescription(directory: String, patternId: String): String = {
+    // This method trims the description to a max length of 500.
+    // The basic description has a limit of 500 chars, so it is required to trim if it is longer than that
+    // The trim is being done by removing a sentence from the end each time until it has less than 500 length.
+    @tailrec
+    def descriptionTrimAux(description: String): String = {
+      if (description.length < 500) {
+        description
+      } else {
+        descriptionTrimAux(description.substring(0, description.lastIndexOf(".")))
+      }
+    }
+
+    val resourcesFolder = s"$directory/src/main/resources"
+    val messagesFiles =
+      File(resourcesFolder) / directory / "src" / "main" / "resources" / "codenarc-base-messages.properties"
+    val messagesFileContent = messagesFiles.contentAsString
+
+    val descriptionRegexFinder = s"$patternId.description=(.*)".r
+
+    val description = descriptionRegexFinder
+      .findFirstMatchIn(messagesFileContent)
+      .map(_.group(1))
+      .toList
+      .headOption
+      .getOrElse("")
+
+    descriptionTrimAux(description)
+  }
+
+  /**
+    * Get list of all CodeNarc's supported rules from git repos documentation
+    *
+    * @param toolVersion CodeNarc's version
+    * @return
+    */
+  def listRulesSupportedFromGitDocumentation(toolVersion: String): Seq[RulePatternInformation] =
+    GitHelper.withRepository(codeNarcGitRepository, toolVersion)(directory => {
+      val documentationFolder = directory / "docs"
+
+      val documentationStartingPoint = documentationFolder / codeNarcDocumentationStartingPoint
+      val htmlListOfRules = MarkdownHelper.markdownToHtmlXml(documentationStartingPoint)
+
+      // patterns list
+      for {
+        li <- htmlListOfRules \\ "li"
+        a <- li \ "a"
+
+        ruleName = a.text
+        ruleInfoLocation = (a \ "@href").text
+
+        priority = rulePriority(ruleName)
+        parameters = ruleParameters(ruleName)
+
+        ruleInformationMdFile = extractMdFilenameFromHref(ruleInfoLocation)
+        ruleExtendedInfo = getRuleExtendedInfoFromMarkdown(
+          documentationFolder.pathAsString,
+          ruleName,
+          ruleInformationMdFile
+        )
+        ruleBasicDescription = getRuleBasicDescription(directory.toString, ruleName)
+
+        patternSpec = Pattern.Specification(
+          Pattern.Id(ruleName),
+          levelFromPriority(priority),
```

Comment:
Here you can pass only `priority` field:
`priority.map(levelFromPriority(_.priority))`

Reply:
Done